binderhub/builder.py

```diff
@@ -250,8 +250,26 @@ async def get(self, provider_prefix, _unescaped_spec):
         except Exception as e:
             await self.fail("Error resolving ref for %s: %s" % (key, e))
             return
+
         if ref is None:
-            await self.fail("Could not resolve ref for %s. Double check your URL." % key)
+            error_message = ["Could not resolve ref for %s. Double check your URL." % key]
+
+            split_key = key.split("/")
+            repo_provider = split_key[0].split(":")[0]
+            resolved_branch = split_key[-1]
+
+            if repo_provider == "gh":
+                error_message.append('GitHub recently changed default branches from "master" to "main".')
+
+                if resolved_branch == "master":
```

Comment:
resolved_branch should be available as `provider.unresolved_ref` so we shouldn't need to parse out the request string